packages/near-api-js/src/account.ts

```diff
@@ -482,7 +482,26 @@ export class Account {
      * @param options.blockQuery specifies which block to query state at. By default returns last "optimistic" block (i.e. not necessarily finalized).
      * @returns {Promise<any>}
      */
-    async viewFunction(
+
+    async viewFunction(obj : any, ...restArgs: any[]) {
```

Comment:
For the types I think we should reuse the [FunctionCallOptions](https://github.com/near/near-api-js/commit/a05667361ddab208423f5a5c0e3dd0ce182b9880#diff-ae151444b91b7e5e0a6aa753024f7900f0ff0447afaa2d18ad6619d5c3e12318R81) type. 

The best way to do this imo is to have a general `FunctionCallOptions` type, remove `walletMeta` and `walletCallbackUrl` and then create a `ChangeFunctionCallOptions` type and a `ViewFunctionCallOptions` type since everything in the `options` arg of `viewFunction` is in `FunctionCallOptions` except for `parse` and `blockQuery`:

```typescript
// using the same FunctionCallOptions with walletCallbackUrl and walletMeta removed
type ChangeFunctionCallOptions = FunctionCallOptions & { walletMeta?: string; walletCallbackUrl?: string; };
type ViewFunctionCallOptions = FunctionCallOptions & { parse?: (response: Uint8Array) => any; blockQuery?: BlockReference; };
```
We just need to maintain the same default arg logic for all args as currently in the `viewFunction` block, maintain the jsdoc we have for `walletMeta` and `walletCallbackUrl` and change `Account.functionCall` to use the `ChangeFunctionCallOptions` type. After this is done `viewFunctionV2` can have the `ViewFunctionCallOptions` type and v1 can have the same type signature as currently. Wdyt?

Reply:
Thank you for amazing review, I totally agree with all the suggestions.